### PR TITLE
refactor: enabled by default

### PR DESCRIPTION
**Pull request by SukkaW**
Enable the plugin by default.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,14 +2,14 @@
 
 'use strict';
 
-if (hexo.config.nofollow && hexo.config.nofollow.enable) {
-  const config = hexo.config.nofollow;
+if (hexo.config.nofollow.enable === false) return;
```
[the remark this answers is not in the document]
Comment (SukkaW) [reply index.js]: Rebased and fixed

File: index.js [hunk 2]
```diff
@@ -2,14 +2,14 @@
...
 
-  if (!config.field) config.field = 'site';
+if (!config.field) config.field = 'site';
 
-  if (config.field === 'post') {
-    hexo.extend.filter.register('after_post_render', require('./lib/filter'));
-  } else {
-    hexo.extend.filter.register('after_render:html', require('./lib/filter'));
-  }
+if (config.field === 'post') {
```
Comment (curbengh) [inline index.js]: should have `if (config.enable === true)` before this line.

Comment (SukkaW) [reply index.js]: Forgot it. `if (!config.enable) return;` is now added.

File: index.js [hunk 3]
```diff
@@ -2,14 +2,16 @@
 
 'use strict';
 
-if (hexo.config.nofollow && hexo.config.nofollow.enable) {
-  const config = hexo.config.nofollow;
+const config = Object.assign({
```
Comment (curbengh) [inline index.js]: you should initialize all variables, currently there is error from
https://github.com/hexojs/hexo-filter-nofollow/blob/6ca54a2ac4ca600c8a58bee1b1c30b4fc9899288/lib/filter.js#L48
`Cannot read property field of undefined`.

This is how I recently do it in my plugin:
https://github.com/curbengh/hexo-yam/blob/master/index.js

Comment (SukkaW) [reply index.js]: A default `field` and `exclude` config is now included in `Object.assign`.

File: index.js [hunk 4]
```diff
@@ -2,14 +2,24 @@
...
+  enable: true,
+  field: 'site',
+  exclude: []
+}, hexo.config.nofollow);
 
-  if (!config.field) config.field = 'site';
+let exclude = hexo.config.nofollow.exclude;
+if (exclude && !Array.isArray(exclude)) {
+  exclude = [exclude];
+  hexo.config.nofollow.exclude = exclude;
```
Comment (curbengh) [inline index.js]: Suggest doing array conversion in `lib/filter.js` without changing user config.

File: lib/filter.js [hunk 5]
```diff
@@ -25,13 +23,19 @@ function isExternal(url, config) {
 
 module.exports = function(data) {
   const hexo = this;
-  const config = hexo.config;
+  let config = hexo.config;
+
+  let exclude = config.nofollow.exclude;
+  if (exclude && !Array.isArray(exclude)) {
+    exclude = [exclude];
+    config.nofollow.exclude = exclude;
```
Comment (curbengh) [inline lib/filter.js]: `config.nofollow.exclude = [exclude];`?

File: lib/filter.js [hunk 6]
```diff
@@ -25,13 +23,19 @@ function isExternal(url, config) {
 
 module.exports = function(data) {
   const hexo = this;
-  const config = hexo.config;
+  let config = hexo.config;
```
Comment (curbengh) [inline lib/filter.js]: can be const, tested it.

**Pull request merged**